lib/src/main/java/org/esbtools/eventhandler/PollingDocumentEventProcessorRoute.java

```diff
@@ -31,34 +27,43 @@
 import java.util.Map;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicInteger;
 
+import org.apache.camel.builder.RouteBuilder;
+
+import com.google.common.collect.Iterables;
+import com.google.common.util.concurrent.Futures;
+
 public class PollingDocumentEventProcessorRoute extends RouteBuilder {
     private final DocumentEventRepository documentEventRepository;
     private final Duration pollingInterval;
     private final int batchSize;
+    private final Duration documentEventProcessTimeout;
     private final String documentEndpoint;
     private final String failureEndpoint;
     private final String routeId;
 
     private static final AtomicInteger idCounter = new AtomicInteger(1);
 
     public PollingDocumentEventProcessorRoute(DocumentEventRepository documentEventRepository,
-            Duration pollingInterval, int batchSize, String documentEndpoint,
+            Duration pollingInterval,  Duration documentEventProcessTimeout,int batchSize, String documentEndpoint,
             String failureEndpoint) {
         this.documentEventRepository = documentEventRepository;
         this.pollingInterval = pollingInterval;
+        this.documentEventProcessTimeout = documentEventProcessTimeout;
         this.batchSize = batchSize;
         this.documentEndpoint = documentEndpoint;
         this.failureEndpoint = failureEndpoint;
         this.routeId = "documentEventProcessor-" + idCounter.getAndIncrement();
     }
 
     public PollingDocumentEventProcessorRoute(DocumentEventRepository documentEventRepository,
-    		Duration pollingInterval, int batchSize, String documentEndpoint,
+    		Duration pollingInterval, Duration documentEventProcessTimeout, int batchSize, String documentEndpoint,
             String failureEndpoint, String routeId) {
         this.documentEventRepository = documentEventRepository;
         this.pollingInterval = pollingInterval;
+        this.documentEventProcessTimeout = documentEventProcessTimeout;
```

Comment:
For brevity I would consider naming these parameters and fields all just `processTimeout`. Prefixing "documentEvent" might be redundant.